src/transformers/integrations/executorch.py

```diff
@@ -206,13 +207,23 @@ def convert_and_export_with_cache(
             example_cache_position if example_cache_position is not None else torch.tensor([0], dtype=torch.long)
         )
 
-        # Due to issue https://github.com/pytorch/pytorch/issues/128394, we need to switch to use an internal
-        # export API and pre_dispatch=False. Switch to use the public API once the issue is included in 2.5 release.
-        exported_program = torch.export._trace._export(
-            TorchExportableModuleWithStaticCache(model),
-            args=(example_input_ids,),
-            kwargs={"cache_position": example_cache_position},
-            pre_dispatch=False,
-            strict=True,
-        )
+        if version.parse(torch.__version__) >= version.parse("2.5.0"):
+            exported_program = torch.export.export(
+                TorchExportableModuleWithStaticCache(model),
+                args=(example_input_ids,),
+                kwargs={"cache_position": example_cache_position},
+                strict=True,
```

Comment:
Use strict=False here.

Reply:
I suppose it's ok to have strict=True for tests unless we experience any difficulties with tests to pass, wdyt?